crates/agglayer-storage/src/stores/state/mod.rs

```diff
@@ -30,21 +33,54 @@ impl StateStore {
 }
 
 impl StateWriter for StateStore {
-    fn insert_certificate_header(&self, certificate: &Certificate) -> Result<(), Error> {
-        self.db.put::<CertificatePerNetworkColumn>(
-            &certificate_per_network::Key {
-                network_id: *certificate.network_id,
-                height: certificate.height,
-            },
+    fn insert_certificate_header(
+        &self,
+        certificate: &Certificate,
+        status: CertificateStatus,
+    ) -> Result<(), Error> {
+        // TODO: make it a batch write
+        self.db.put::<CertificateHeaderColumn>(
+            &certificate.hash(),
             &CertificateHeader {
                 certificate_id: certificate.hash(),
                 network_id: certificate.network_id,
                 height: certificate.height,
                 epoch_number: None,
                 certificate_index: None,
-                new_local_exit_root: certificate.new_local_exit_root,
+                new_local_exit_root: certificate.new_local_exit_root.into(),
+                status: status.clone(),
             },
-        )
+        )?;
+
+        if let CertificateStatus::Settled = status {
```

Comment:
in this case the fields `epoch_number` and `certificate_index` are not written? or will it be updated afterwards?

Reply:
Will be done when we tackle the settlement on L1